pkg/cmd/issue/view/issuePrintFormatter_test.go

```diff
@@ -0,0 +1,619 @@
+
+			createdAtTime, err := time.Parse(time.RFC3339, tc.createdAt)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			issue := &api.Issue{
+				Title:  tc.title,
+				Number: tc.number,
+				Comments: api.Comments{
+					TotalCount: 1,
+				},
+				State:     tc.state,
+				CreatedAt: createdAtTime,
+				Author: api.Author{
+					Login: tc.author,
+				},
+			}
+
+			ipf := NewIssuePrintFormatter(issue, ios, createdAtTime.AddDate(0, 0, 1), tc.baseRepo)
+			ipf.header()
+			assert.Equal(t, stdout.String(), tc.expected)
+		})
+	}
+}
+
+func Test_reactions(t *testing.T) {
+	tests := map[string]struct {
+		reactions api.ReactionGroups
+		expected  string
+	}{
+		"no reactions": {
+			reactions: api.ReactionGroups{},
+			expected:  "",
+		},
+		"single thumbs up reaction": {
+			reactions: api.ReactionGroups{
+				api.ReactionGroup{
+					Content: "THUMBS_UP",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 1,
+					},
+				},
+			},
+			expected: "1 \U0001f44d\n",
+		},
+		"all reactions": {
+			reactions: api.ReactionGroups{
+				api.ReactionGroup{
+					Content: "THUMBS_UP",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 1,
+					},
+				},
+				api.ReactionGroup{
+					Content: "THUMBS_DOWN",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 2,
+					},
+				},
+				api.ReactionGroup{
+					Content: "LAUGH",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 3,
+					},
+				},
+				api.ReactionGroup{
+					Content: "HOORAY",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 4,
+					},
+				},
+				api.ReactionGroup{
+					Content: "CONFUSED",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 5,
+					},
+				},
+				api.ReactionGroup{
+					Content: "HEART",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 6,
+					},
+				},
+				api.ReactionGroup{
+					Content: "ROCKET",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 7,
+					},
+				},
+				api.ReactionGroup{
+					Content: "EYES",
+					Users: api.ReactionGroupUsers{
+						TotalCount: 8,
+					},
+				},
+			},
+			expected: "1 \U0001f44d • 2 \U0001f44e • 3 \U0001f604 • 4 \U0001f389 • 5 \U0001f615 • 6 \u2764\ufe0f • 7 \U0001f680 • 8 \U0001f440\n",
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			ios, _, stdout, _ := iostreams.Test()
+			ios.SetStdoutTTY(true)
+			ios.SetStdinTTY(true)
+			ios.SetStderrTTY(true)
+
+			timeNow, err := time.Parse(time.RFC3339, "2022-01-01T00:00:00Z")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			issue := &api.Issue{
+				ReactionGroups: tc.reactions,
+			}
+
+			ipf := NewIssuePrintFormatter(issue, ios, timeNow, ghrepo.New("OWNER", "REPO"))
+			ipf.reactions()
+			assert.Equal(t, stdout.String(), tc.expected)
+		})
+	}
+}
+
+// Note: this is currently over-testing. We should be able to mock the
+// return from GetAssigneeListString (which has its own tests), but I think
+// that requires a larger refactor of the Issue code to leverage interfaces
+// for mocking.
```

Comment:
@williammartin I'm struggling a bit to wrap my head around how exactly to leverage an interface with `Label` to allow for mocking of this `GetAssigneeListString` method in these tests... Any insights would be appreciated

Reply:
Riiiiight. I think we're feeling friction here not due to a missing interface but due to an overloading of the `api` domain.

Putting this PR mentally aside for a moment, if I asked you "How does converting an issue's assignees into a string relate to the GitHub API?" I think you would probably say "what?". Or if I said to you, "Is the API responsible for how data is presented?", I think you would probably say "No". These methods `GetAssigneeListString` and `GetProjectListString` and `SortAlphabeticallyIgnoreCase` are presentation logic that this PR is hanging from the API domain type.

If I ask myself "what does the `IssuePrintFormatter do right now?`, I'd say "Well, it takes an issue that is returned from the API, converts that data into a format for presentation, then does some conditional printing of it". Which makes me suspicious that the conversion is a responsibility that should live elsewhere.

My gut feeling here is that we're missing a new domain type in the `view` package that accurately captures what it means to view an issue. One of the fields on that would be `assignees string`, containing the already stringified data.

I would expect a function (naming TBD) like:

```go
func apiIssueToPresentationIssue(api.Issue) presentationIssue {
    // here we can test that assignees are turned to strings correctly
}
```

Then the `IssuePrintFormatter` doesn't have anything to mock, it already has the plain old data it needs.

Let's chat about this more if it's not clear.

Reply:
Ah, I think I'm following... You're right, that we're overloading the API. This is an excellent idea that will help untangle this further. Thanks!

Reply:
Alright, I took a stab at this by introducing the `presentationIssue` interface. It did remove the need for mocking 🙌 but I'm sure there's more I'm missing. Let's chat about it in our pairing sesh.